app/jobs/populate_register_data_in_db_job.rb

```diff
@@ -0,0 +1,117 @@
+class PopulateRegisterDataInDbJob < ApplicationJob
+  queue_as :default
+
+  def initialize
+    @registers_client ||= RegistersClient::RegisterClientManager.new(cache_duration: 60)
+  end
+
+  def bulk_remove_existing_records(register, entry_type, record_keys)
+    unless record_keys.empty?
+      Record.where(spina_register_id: register.id, entry_type: entry_type, key: record_keys).delete_all
+    end
+  end
+
+  def bulk_save(entries, records)
+    Entry.import(entries)
+    Record.import(records)
+  end
+
+  def populate_register(register)
+    register_data = @registers_client.get_register(register.name.parameterize, register.register_phase.downcase)
+    register_data.refresh_data
+    populate_data(register, register_data, 'user')
+    populate_data(register, register_data, 'system')
+
+    register.fields = register_data.get_field_definitions.map { |field| field.item.value['field'] }.join(',')
+    register.save
+  end
+
+  def populate_data(register, register_data, entry_type)
+    entries = []
+    records = []
+    count = 0
+    latest_entry = Entry.where(spina_register_id: register.id, entry_type: entry_type).order(:entry_number).reverse_order.limit(1).first
```

Comment:
Replace `limit(1).first` with `first(1)`  https://apidock.com/rails/ActiveRecord/FinderMethods/first ?

Reply:
Actually there is no need for the argument as it defaults to 1. Good suggestion!